For some reason you need to run powerlevel9k_prepare_prompts once (in a widget) in a session to get this to work
--- michaelxmcbride---zsh-dircycle/dircycle.zsh	2020-05-30 02:47:30.172866443 -0400
+++ michaelxmcbride---zsh-dircycle/dircycle.zsh-patch	2020-05-30 02:47:27.522927038 -0400
@@ -18,7 +18,9 @@
         prompt_pure_async_tasks
         prompt_pure_preprompt_render
     else
-        zle reset-prompt
+        _p9k_precmd
+        zle .reset-prompt
+        zle -R
     fi
 }

